src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts

```diff
@@ -0,0 +1,83 @@
+import { BaseMongoRepository } from './base-mongo-repository'
+import { VideoMixerDeviceRepository } from '../interfaces/video-mixer-device-repository'
+import { MongoDatabase } from './mongo-database'
+import { VideoMixerConfiguration } from '../../../model/value-objects/video-mixer-configuration'
+import { NotFoundException } from '../../../model/exceptions/not-found-exception'
+import { DeviceEventEmitter } from '../../../business-logic/services/interfaces/device-event-emitter'
+import { ChangeStream, ChangeStreamDocument, ChangeStreamOptions } from 'mongodb'
+import { MongoChangeEvent } from './mongo-enums'
+
+interface MongoDevice {
+  _id: string
+  name: string
+  settings: {
+    devices: Record<string, {
+      options: {
+        host: string
+        port: number
+      }
+    }>
+  }
+}
+
+const COLLECTION_NAME: string = 'peripheralDevices'
+
+const PLAYOUT_GATEWAY_POSTFIX: string = 'PlayoutCoreParent'
+const VIDEO_MIXER_NAME: string = 'atem'
+
+/**
+ * This Repository only exist to fetch the hostname and port of the configured VideoMixer (currently hardcoded to Atem) while device settings resides in Sofie.
+ * Once proper device settings is introduced in Alba. This SHOULD be deleted!
+ */
+export class MongoVideoMixerDeviceRepository extends BaseMongoRepository<MongoDevice> implements VideoMixerDeviceRepository {
+
+  constructor(mongoDatabase: MongoDatabase, private readonly deviceEventEmitter: DeviceEventEmitter) {
+    super(mongoDatabase)
+    mongoDatabase.onConnect(COLLECTION_NAME, () => this.listenForVideoMixerChanges())
+  }
+
+  private listenForVideoMixerChanges(): void {
+    const options: ChangeStreamOptions = { fullDocument: 'updateLookup' }
+    const changeStream: ChangeStream = this.getCollection().watch<MongoDevice, ChangeStreamDocument<MongoDevice>>([], options)
+    changeStream.on('change', (change: ChangeStreamDocument<MongoDevice>) => {
+      switch (change.operationType) {
+        case MongoChangeEvent.UPDATE: {
+          const mongoDevice: MongoDevice | undefined = change.fullDocument
+          if (!mongoDevice) {
+            return
+          }
+          const videoMixerConfiguration: VideoMixerConfiguration = this.findVideoMixerConfiguration(mongoDevice)
+          this.deviceEventEmitter.emitVideoMixerConfigurationUpdated(videoMixerConfiguration)
+          break
+        }
+        default: {
+          // Ignore the rest
+        }
+      }
+    })
+  }
+
+  private findVideoMixerConfiguration(mongoDevice: MongoDevice): VideoMixerConfiguration {
+    const key: string | undefined = Object.keys(mongoDevice.settings.devices).find(key => key.includes(VIDEO_MIXER_NAME))
```

Comment:
From what you are using it for, i would suggest this renaming :)

```suggestion
    const deviceName: string | undefined = Object.keys(mongoDevice.settings.devices).find(key => key.includes(VIDEO_MIXER_NAME))
```

Reply:
Updated